appIcons.js

```diff
@@ -271,15 +271,18 @@ const MyAppIcon = new Lang.Class({
             case clickAction.MINIMIZE:
                 // In overview just activate the app, unless the acion is explicitely
                 // requested with a keyboard modifier
-                if (!Main.overview._shown || !modifiers){
-                    if (this.app == focusedApp)
+                if (!Main.overview._shown || modifiers){
+                    // If we have button=2 or Shift, allow minimization even if
+                    // the app is not focused
+                    if (this.app == focusedApp || button == 2
+                        || modifiers & Clutter.ModifierType.SHIFT_MASK)
```

Comment:
Here I added `button == 2 || modifiers & Clutter.ModifierType.SHIFT_MASK` as options in the `if`. Not sure if you actually want this, but this is what the behaviour was like before: when NOT using the regular click, one can minimize windows even if they are not focused.

Below I just adjust the indentation to be consistent with the rest of the code.